syntax = "proto2";
package st.planning;
// option go_package = "st.proto.ai/onboard/planning.router/proto/route_manager_output";

import "modules/cnoa_pnc/planning/proto/lite_common.proto";
import "modules/cnoa_pnc/planning/proto/turn_signal.proto";
import "modules/cnoa_pnc/planning/proto/route.proto";

message AlternativePathProto {
  optional RouteSectionSequenceProto route_sections_from_current = 1;
  optional RouteNaviInfoProto route_navi_info = 2;
  optional double roundabout_distance = 3;
}

message RouteManagerOutputProto {
  optional LiteHeader header = 1;

  // NOTE: 'route' and 'route from current' are deprecated,
  // reserve for compatibility.
  // optional RouteProto route = 2;
  // optional RouteProto route_from_current = 3;
  // optional RouteContentProto route_content = 4 [deprecated = true];

  optional RouteSectionSequenceProto route_sections_from_current = 6;
  repeated uint64 avoid_lanes = 13;
  optional RouteNaviInfoProto route_navi_info = 15;

  optional TurnSignal signal = 7;
  optional bool rerouted = 8;
  optional bool is_bus = 9 [deprecated = true];

  // optional MultipleStopsRequestProto.StopProto destination_stop = 10;
  // optional RoutingRequestProto routing_request = 14;

  optional int64 update_id = 11;
  optional bool reset = 12;

  optional double recommend_lane_speed_limit = 16;  // m/s

  optional AlternativePathProto alternative_path_proto = 17;

  optional bool is_valid = 18;

  // NEXT ID: 19.
  reserved 5;
}